### ISSUE TITLE
Path imported by `[<ReactComponent(import="...", from="...")>] should not be relative to the caller side

### ISSUE THREAD

**Issue by MangelMaxime** [fable-hub/Feliz#624]
I have a `Code.tsx` file with a `Components.fs` file where I am binding the component for usage in F#:

```fs
[<ReactComponent(import="default", from="./${outDir}/Code.tsx")>]
let Code (code: string) (language : string) = React.imported()
```

Then in another file from another directory, I am calling the F# `Code` function to use the component. The generated code is:

```fs
import Code from "./Code.tsx";
```

The issue is that `./Code.tsx` is invalid because the caller file is not inside of the same directory as `Components.fs`:

```text
.
├── Components
│   ├── Code.tsx
│   ├── Components.fs
│   └── Components.fs.js
├── Utils
│   ├── Html.fs
```

The import path should be relative to where the API is defined.

IHMO this should be the default behaviour otherwise, it means that if you need to use a JSX component then you need to re-declare it several times 😅

A workaround, is to create a standard Feliz binding using `Interop.reactApi.createElement` to we can use Fable special macro`${outDir}`:

```fs
type Components =

    static member Code (code: string) (language: string) =
        Interop.reactApi.createElement (
            import "default" "${outDir}/Code.tsx",
            {|
                code = code
                language = language
            |}
        )
```

It is important to not mark this member as `inline` otherwise it will not work because the call site is not stable.

**Comment by Zaid-Ajaj** [comment]
Hi there @MangelMaxime so the problem here is that we need access to the value of `outDir` from the compiler. Here is where use it:
```fs
let makeImport (selector: string) (path: string) =
    Fable.Import({ Selector = selector.Trim()
                   Path = path.Trim()
                   Kind = Fable.UserImport(false) }, Fable.Any, None)
```
Maybe Fable compiler should automatically replace `${outDir}` with its value when encountering `Fable.Import` AST node? 🤔 

OR Fable could expose the `outDir` as an environment variable such that we get access to it in compile-time:
```fs
let path = 
  match env "FABLE_OUT_DIR" with
  | Some outDir -> path.Replace("${outDir}", outDir).Trim()
  | None -> path.Trim()
```

**Comment by MangelMaxime** [comment]
Replacing `${outDir}` automatically was my idea but I need to check when it happens in Fable.

One reason to do it this way, is like that plugins don't know to worry about path and we don't have to duplicate the logic once per plugin.

**Comment by Zaid-Ajaj** [comment]
> One reason to do it this way, is like that plugins don't know to worry about path and we don't have to duplicate the logic once per plugin.

Yeah it makes more sense to do that in the compiler ✅ you probably need to traverse the resulting AST from compiler plugins, check for `Fable.Import` nodes and update their `Path` if it contains the`${outDir}` macro

**Comment by pradeepr-roboticist** [comment]
Thanks for all the work on Feliz. The dev experience has been great!

I am hitting this issue right now. I tried the `${outDir}` approach but it did not work. Has there been any work towards this? Thanks!

**Comment by Freymaurer** [comment]
Hey @pradeepr-roboticist! I am afraid there is currently no full solution for this yet.

Here is a workaround until then!

```fsharp
[<Erase; Mangle(false)>]
type MyJSX = 

    [<ReactComponent>]
    static member Main() : ReactElement = Fable.Core.JsInterop.importDefault "./MyJSX.jsx"
```

This is not a true binding, as it creates minimal additional js code, but because the import is handled locally in the file, you can import Main anywhere you like

```js
import MyJSX from "./MyJSX.jsx";

export const Main = MyJSX;
```

I didnt encounter this issue yet, this should be adressed correctly.

**Comment by Freymaurer** [comment]
Something that might enhance your experience with fsharp and feliz. The `[<ReactComponent>]` attribute actually transforms your arguments into a javascript object.

So for a jsx components like this:

```jsx
import React from 'react';

export default function MyJSX({init}) {
    const [count, setCount] = React.useState(init);
    return (
        <div>
            <h1>Hello, World!</h1>
            <p>This is a simple JSX component.</p>
            <button onClick={() => setCount(count + 1)}>
                Click me! Count: {count}
            </button>
        </div>
    );
}
```

It expects a javascript object with a field called `init`. You can use the following fsharp syntax:

```fsharp
[<Erase; Mangle(false)>] // This just removes any references to the type in the js file and makes it cleaner/smaller
type MyJSX = 

    [<ReactComponent>]
    static member Main(init: int) : ReactElement = Fable.Core.JsInterop.importDefault "./MyJSX.jsx"
```

Here the `init: int` will be transformed to be part of a js object. The same as `{|init: int|}`. But this syntax has a major upside, using optional parameters is extremely nice to do.

```jsx
export default function MyJSX({init, optional1, optional2}) {
```

Can be done like this:

```fsharp
[<Erase; Mangle(false)>] // This just removes any references to the type in the js file and makes it cleaner/smaller
type MyJSX = 

    [<ReactComponent>]
    static member Main(init: int, ?optional1, ?optional2) : ReactElement = Fable.Core.JsInterop.importDefault "./MyJSX.jsx"

// and called like this

MyJSX.Main(42)

MyJSX.Main(42, "Hello")

MyJSX.Main(42, "Hello", "World")

//or even

MyJSX.Main(42, optional2 = "World")
```

**Comment by Freymaurer** [comment]
Hey! I had some time to look into this issue and added an additional layer of path resolving in the Feliz.CompilerPlugins in #682.

Now paths are resolved between the fsharp source file with the `[<ReactComponent(import,from)>]` attribute and the fsharo source file calling this binding.

I also added several test which should ensure correct behavior. :sparkles:
